lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -839,6 +893,168 @@ struct ConvertElementwiseOp : ConversionPattern {
+    Value ceilModeFalse =
+        rewriter.create<CmpIOp>(loc, CmpIPredicate::eq, ceilMode, falseValue);
+    rewriter.create<AssertOp>(
+        loc, ceilModeFalse,
+        rewriter.getStringAttr("only ceil_mode false is supported"));
+
+    SmallVector<int64_t, 4> paddingIncludingNC = {0, 0};
+    paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
+                              paddingInts.end());
+    Value paddedInput = getPaddedTensor(op, rewriter, self, paddingIncludingNC);
+
+    Value N = getDimOp(rewriter, loc, self, 0);
+    Value C = getDimOp(rewriter, loc, self, 1);
+    Value H = getDimOp(rewriter, loc, self, 2);
+    Value W = getDimOp(rewriter, loc, self, 3);
+
+    SmallVector<Value> paddingIntValues =
+        getAsConstantIntValues(rewriter, loc, paddingInts);
+    SmallVector<Value> dilationIntValues =
+        getAsConstantIntValues(rewriter, loc, dilationInts);
+    SmallVector<Value> kernelSizeIntValues =
+        getAsConstantIntValues(rewriter, loc, kernelSizeInts);
+    SmallVector<Value> strideIntValues =
+        getAsConstantIntValues(rewriter, loc, strideInts);
+
+    Value Hout = getOutputDimForConvOps(
+        rewriter, loc, H, paddingIntValues[0], dilationIntValues[0],
+        kernelSizeIntValues[0], strideIntValues[0]);
+    Value Wout = getOutputDimForConvOps(
+        rewriter, loc, W, paddingIntValues[1], dilationIntValues[1],
+        kernelSizeIntValues[1], strideIntValues[1]);
+
+    // Initialize output tensor with smallest floating point value
+    Value outTensor = rewriter.create<linalg::InitTensorOp>(
+        loc, ValueRange{N, C, Hout, Wout}, elementType);
+    auto initialAttr = rewriter.getFloatAttr(
+        elementType,
+        APFloat::getSmallest(
+            elementType.cast<mlir::FloatType>().getFloatSemantics(),
+            /*Negative*/ true));
+    Value initValue = rewriter.create<ConstantOp>(loc, initialAttr);
+    Value outTensorInitialized =
+        rewriter.create<linalg::FillOp>(loc, initValue, outTensor).getResult(0);
+
+    auto stridesAttr = rewriter.getI64VectorAttr(strideInts);
+    auto dilationAttr = rewriter.getI64VectorAttr(dilationInts);
+    Value windowTensor = rewriter.create<linalg::InitTensorOp>(
+        loc, getAsConstantIndexValues(rewriter, loc, kernelSizeInts),
+        elementType);
+
+    Value maxPool2d = rewriter
+                          .create<linalg::PoolingNchwMaxOp>(
+                              loc, outTensorInitialized.getType(),
+                              ValueRange{paddedInput, windowTensor},
+                              outTensorInitialized, stridesAttr, dilationAttr)
+                          .getResult(0);
+    Type newResultType = getTypeConverter()->convertType(op.getType());
+    rewriter.replaceOpWithNewOp<tensor::CastOp>(op, newResultType, maxPool2d);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class ConvertAtenFlattenUsingIntsOp
```

Comment:
this one is pretty tricky. can you add tests for it? (see tests for unsqueeze)

Reply:
thanks for the suggestion. It does catch an error. Changed RefineTypes for flattening 0 rank case to set the resulting dim to be rank 1 with size 1.

Reply:
Thanks. I was also referring to a .mlir file like test/Conversion/TorchToLinalg/unsqueeze.mlir which really helps isolate the lowering.

Reply:
some how I missed those .mlir tests. Added suggested.